lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+```
+
+```{r, echo=FALSE, eval=FALSE}
+# Solution
+fit_challenge2 <- glm(Disease ~ weight + sex + hindfoot_length, 
+                      data = challenge2, family = binomial)
+summary(fit_challenge2)
+tidy(fit_challenge2)
+-1.035 + (-0.0003 * 50) + (-0.0338 * 0) + (-0.0004 * 30)
+```
+
+### Logistic regression
+
+Logistic regression is a technique that has the $y$ as a binary or categorical
+variable (e.g. female vs male, mammal vs plant vs bacteria). We won't be
+covering this in too much detail, but I will go over it to highlight the
+flexibility that GLM has in handling different types of data.
+
+The reason why the result of the challenge two above was difficult to interpret
+was because there's a bit more to logistic regression then with linear
+regression. It becomes clearer when we look at the formula:
+
+$$ logit(p) = \ln\left(\frac{P(y=1)}{P(y=0)}\right) = a + Xb + e$$
+
+In this case, the $y$ is the logged odds of the event occurring (in the
+challenge the event was disease). In order to interpret the model results, we
+need to exponentiate both sides of the equation.
+
+$$ \exp\left(\ln\left(\frac{p}{1 - p}\right)\right) = \exp(a + Xb + e)$$
+$$ \frac{p}{1 - p} = e^{a + Xb + e}$$
+
+where $p$ is the probability of the event (equal to 1, in this case of disease).
+So if we exponentiate the estimates, and solve for the formula, we can interpret
+the result as the odds of an event occurring.
+
+```{r}
+exp(-1.035 + (-0.0003 * 50) + (-0.0338 * 0) + (-0.0004 * 30))
+```
+
+In this case, for a female with a weight of 50 and hindfoot of 30, the odds of
+disease is 0.345, fairly small. Interpretation is actually a bit more
+complicated than that. So depending on your final project, if you need/want to
+use this, we can help walk you through this more.
+
+## Confidence intervals, and p-values
+
+So what's the point of using `glm`? Not only does it create a model with
+constants and coefficients, but it is also used to calculate how *certain* we
+are about the results. The main power of GLM comes from the ability to estimate
+model parameters and derive meaning from how certain $x$ variables (independent
+variables) influence the $y$ (dependent variable). With his model we can use to
+it *predict* what could happen to some value if we knew or could change the
+independent variables ($x$). But we also want to know whether this model
+reflects reality.
+
+From GLM we can determine the *confidence* (or rather, the *uncertainty*) we
+have about the beta ($b$) estimates. In the case of linear regression model, we
+can use `tidy` with the `conf.int` argument to calculate the confidence
+interval.
+
+```{r}
+fit_hf <- glm(weight ~ hindfoot_length, data = portal)
+tidy(fit_hf, conf.int = TRUE) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+In this case, the uncertainty of the estimate for hindfoot length is between
+2.524 to 2.58, which is pretty narrow (that's a good thing). This tells us that
```

Comment:
Maybe mention which confidence interval you're calculating (i.e. 90%, 95%) and what that means, like 95% means 95% of the probability for that parameter is within that interval.